doc/channel-plugin.md

```diff
@@ -0,0 +1,58 @@
+## Channel Plugin
+
+Channel plugins are processes that run independently of each other.
+
+Channel plugin uses stdin and stdout as json format to communicate. The request and response structure is inspired by
+JSON-RPC.
+
+### Request
+
+The request must be in json format and should contain following keys:
+
+- `method`: The request method to call
+- `params`: The params for request method
+- `id`: Unsigned int value. Require to assign the response to it's request as responses can be out of order
+
+Example:
+
+```
+{
+    "method": "Add",
+    "params": {
+        "num1": 5,
+        "num2": 3
+    },
+    "id" : 2020
+}
+```
+
+The example json files can be seen under `internal/channel/examples/`.
+
+### Response
+
+The response is in json format and contains following keys:
+
+- `result`: The result as json format, omitted if an error has occurred or the method does not return (e.g. setter calls)
```

Comment:
does not return **a value**

It does return, otherwise it wouldn't send a response.